pkg/actor/initialize.go

```diff
@@ -127,9 +128,32 @@ func (init initialize) Act(ctx context.Context, cluster *resource.Cluster) error
 		return errors.Wrap(err, msg)
 	}
 
-	cluster.SetTrue(api.InitializedCondition)
+	fetcher := resource.NewKubeFetcher(ctx, cluster.Namespace(), init.client)
```

Comment:
This is pretty important code, should we add some code-level comments?

Reply:
Done, let me know if you'd like anything else in particular clarified with a comment